crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -427,4 +544,32 @@ pub mod error {
         #[error("invalid solver competition data fetched from database: {0}")]
         InvalidSolverCompetition(anyhow::Error),
     }
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Settlement {
```

Comment:
The error variants seem to have very little context so it's unclear to me from reading the type when it's appropriate to use this error. 🤔

Reply:
deleted